### PR TITLE
Thread rolling logger udf fix

### PR DESCRIPTION
**Pull request by naddeoa**
Apparently the logger layer is responsible for running udfs before processing data with a schema that wants segments. This updates the thread rolling logger's dataset profile abstraction to also run udfs when segments are active.

### CODE REVIEW

File: python/pyproject.toml [hunk 1]
```diff
@@ -275,3 +277,22 @@ typeCheckingMode = "strict"
...
 reportMissingTypeArgumet = false
+
+[tool.ruff]
+line-length = 140
+indent-width = 4
+include = ["whylogs/api/logger/experimental/logger/**/*.py"]
+select = ["E", "F", "I", "W"]
+
+[tool.ruff.isort]
+known-first-party = ["whylogs", "langkit"]
```
Comment (jamie256) [inline python/pyproject.toml]: Do we ever reference langkit in whylogs? I think it would be a circular dependency.

**Review by jamie256 — DISMISSED** [review]
Looks fine to me.

File: python/pyproject.toml (hunk 1 above)
Comment (naddeoa) [reply python/pyproject.toml]: I'll get rid, was copy pasta

**Review by jamie256 — APPROVED** [review]
LGTM!

**Pull request merged**